src/pages/index.tsx

```diff
@@ -1,148 +1,178 @@
-	ctx.fill();
-    };
-
-    const handleCanvasClick = (e: React.MouseEvent<HTMLCanvasElement>) => {
-	const rect = canvasRef.current!.getBoundingClientRect();
-	const x = e.clientX - rect.left;
-	const y = e.clientY - rect.top;
-	setX(x);
-	setY(y);
-
-	const ctx = canvasRef.current!.getContext('2d')!;
-	const img = new Image();
-	img.src = image!;
-
-	ctx.drawImage(img, 0, 0);
-	drawDot(x, y);
-    };
-
-
-    const handleSubmit = async (e: React.FormEvent<HTMLFormElement>) => {
-	e.preventDefault();
-
-	setIsLoading(true);
-
-	if (!image) {
-	    alert('Please upload an image.');
-	    return;
-	}
-
-	try {
-	    const response = await axios.post('/api/edit', { image, x, y, prompt, filename });
-	    setImageUrls(response.data.files);
-	} catch (error) {
-	    console.error('Error submitting the form', error);
-	} finally {
-	    setIsLoading(false)
-	}
-    };
+import Card from "@/components/Card";
+import ImageSpot, { ImageSpotPosition } from "@/components/ImageSpot";
 
-    return (
-      <div className={`container mx-auto px-4 py-8 ${isLoading ? 'opacity-50 pointer-events-none' : ''}`}>
-        <h1 className="text-4xl font-bold mb-8">Edit Image</h1>
-        <form onSubmit={handleSubmit} className="space-y-4">
-            <div>
-                <label className="block text-lg font-medium">Image:</label>
-                <input type="file" onChange={handleImageUpload} className="mt-1" />
-            </div>
-            <div>
-                <label className="block text-lg font-medium">Prompt:</label>
+const Home = () => {
+  const [selectedImage, setSelectedImage] = useState<string | null>(null);
+  const [imageSize, setImageSize] = useState({ height: 0, width: 0 });
+  const [position, setPosition] = useState<ImageSpotPosition | undefined>(
+    undefined
+  );
+  const [prompt, setPrompt] = useState("");
+  const [filename, setFilename] = useState("");
+  const [imageUrls, setImageUrls] = useState<string[]>([]);
+  const [isLoading, setIsLoading] = useState(false);
+
+  const handleImageSelected = (e: React.ChangeEvent<HTMLInputElement>) => {
+    if (e.target.files && e.target.files[0]) {
+      const file = e.target.files[0];
+      const imageUrl = URL.createObjectURL(file);
+
+      const image = new Image();
+      image.src = imageUrl;
+      image.onload = () => {
+        setFilename(file.name);
+        setImageSize({ width: image.width, height: image.height });
+        setSelectedImage(imageUrl);
+      };
+    }
+  };
+
+  const handleImageClick = (position: ImageSpotPosition) => {
+    setPosition(position);
+  };
+
+  const wait = (ms: number) =>
+    new Promise((resolve) => setTimeout(resolve, ms));
+
+  const handleSubmit = async (e: React.FormEvent<HTMLFormElement>) => {
+    e.preventDefault();
+
+    setIsLoading(true);
+
+    await wait(25000);
+
+    try {
+      const response = await axios.post(e.currentTarget.action, {
+        selectedImage,
+        x: position?.x ?? 0,
+        y: position?.y ?? 0,
```

Comment:
NIT: It would be so awesome if the dot was at the center of the image by default.